astropy/coordinates/earth.py

```diff
@@ -265,15 +265,19 @@ def from_geocentric(cls, x, y, z, unit=None):
         if unit.physical_type != "length":
             raise u.UnitsError("Geocentric coordinates should be in units of length.")
 
+        # TODO: this part could be removed, with the try/except around the
+        # assignment to struc["x"], ..., below.  But this is a small API change
+        # in that it will no longer possible to initialize with a unit-full x,
+        # and unit-less y, z. Arguably, though, that would just solve a bug.
```

Comment:
I agree it's a bug.

Reply:
Yes, it annoyed, so I in fact have a follow-up PR ready to fix it! I decided not to apply that here since I felt it was good here to keep the changes to `EarthLocation` as minimal as possible.

Now if only this didn't get messed up by the numpy 1.26 release, which broken our python 3.9 wheels build...